appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java

```diff
@@ -531,7 +568,11 @@ public String getServer() {
   public void clearTicket() {
     ticket = null;
   }
-  
+
+  public boolean isRequestTicket() {
+    return ticket != null;
```

Comment:
I'll make it check isEmpty as well, but I can't easily make getTicket use it otherwise we get an aynchronous race if the ticket is cleared after the isRequestTicket() call but before the return.